src/components/JSONSnippetImportDialog.tsx

```diff
@@ -0,0 +1,204 @@
+import React, { useState } from "react"
+import {
+  Dialog,
+  DialogContent,
+  DialogHeader,
+  DialogTitle,
+  DialogFooter,
+} from "@/components/ui/dialog"
+import { Button } from "@/components/ui/button"
+import { Input } from "@/components/ui/input"
+import { useAxios } from "@/hooks/use-axios"
+import { useToast } from "@/hooks/use-toast"
+import { useLocation } from "wouter"
+import { useGlobalStore } from "@/hooks/use-global-store"
+import { convertCircuitJsonToTscircuit } from "circuit-json-to-tscircuit"
+
+interface JSONSnippetImportDialogProps {
+  open: boolean
+  onOpenChange: (open: boolean) => void
+}
+
+const isValidURL = (url: string) => {
+  try {
+    new URL(url)
+    return true
+  } catch {
+    return false
+  }
+}
+
+export function JSONSnippetImportDialog({
```

Comment:
note that capitalization uses google style, and `JsonSnippet` isn't really a concept we have

```suggestion
export function CircuitJsonImportDialog({
```